python/hsml/core/explicit_provenance.py

```diff
@@ -0,0 +1,338 @@
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.FAULTY,
+            )
+        elif bool(link_json["deleted"]):
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.DELETED,
+            )
+        elif not bool(link_json["accessible"]):
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.INACCESSIBLE,
+                link_json["artifact"]["href"],
+            )
+        elif bool(link_json["accessible"]):
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.NOT_SUPPORTED,
+                link_json["artifact"]["href"],
+            )
+        else:
+            return None
+
+
+class Links:
+    def __init__(self):
+        self._accessible = []
+        self._deleted = []
+        self._inaccessible = []
+        self._faulty = []
+
+    @property
+    def deleted(self):
+        """List of [Artifact objects](../links#artifact) which contains
+        minimal information (name, version) about the entities
+        (feature groups, feature views) they represent.
+        These entities have been removed from the feature store.
+        """
+        return self._deleted
+
+    @property
+    def inaccessible(self):
+        """List of [Artifact objects](../links#artifact) which contains
+        minimal information (name, version) about the entities
+        (feature groups, feature views) they represent.
+        These entities exist in the feature store, however the user
+        does not have access to them anymore.
+        """
+        return self._inaccessible
+
+    @property
+    def accessible(self):
+        """List of [feature groups](../feature_group_api) or
+        [feature views](../feature_view_api) metadata objects
+        which are part of the provenance graph requested. These entities
+        exist in the feature store and the user has access to them.
+        """
+        return self._accessible
+
+    @property
+    def faulty(self):
+        """List of [Artifact objects](../links#artifact) which contains
+        minimal information (name, version) about the entities
+        (feature groups, feature views) they represent.
+        These entities exist in the feature store, however they are corrupted.
```

Comment:
Are the docs correct? No mention of model versions?

Reply:
I updated - it should be feature views and training datasets. We are using provenance to get them from the model. The model is returned in the hsfs version of this explicit_provenance class